buildenv/jenkins/common/build.groovy

```diff
@@ -354,8 +354,19 @@ def upload_artifactory(uploadSpec) {
     // Add BUILD_IDENTIFIER to the buildInfo. The UploadSpec adds it to the Artifact info
     buildInfo.env.filter.addInclude("BUILD_IDENTIFIER")
     buildInfo.env.capture = true
-    server.upload spec: uploadSpec, buildInfo: buildInfo
-    server.publishBuildInfo buildInfo
+
+    //Retry uploading to Artifactory if errors occur
+    def ret = false
+    retry (3) {
+        if (ret) {
+            sleep time: 300, unit: 'SECONDS'
+        } else {
+            ret = true
+        }
+        server.upload spec: uploadSpec, buildInfo: buildInfo
```

Comment:
@AdamBrousseau what happens if the upload occurs twice because the error occurs in publishBuildInfo?

Reply:
Good Morning Peter. Sorry for late reply. We tested the case when a failure happened after uploadSpec. It will do the uploadSpec again and use only the latest one. Thanks! @pshipton